File: ECE39500/exams/bak_midterm-2/midterm-2/PartA/CompilerState.py
import sys

class CompilerState():
    INT_MIN = -sys.maxsize - 1

    def __init__(self):
        return
    
    def comma(self):
        pass

    def value(self):
        pass

    def eol(self):
        pass

    def eof(self):
        pass

    def validOperands(self, operands, count):
        valid = True
        for i in range(count):
            if not valid: break
            if operands[i] == INT_MIN:
                valid = False
        return valid
    
    def printInvalidOperands(self, operands, count):
        for i in range(count):
            if operands[i] < 0:
                print(f"Operand[{i}], {operands[i]} is invalid.")
        return None

File: ECE39500/exams/midterm-2/PartB/src/CompilerState.java
public abstract class CompilerState {
   public abstract void comma( );
   public abstract void value(String str);
   public abstract void eol( );
   public abstract void eof( );


   // check to see if parsed argument values are valid
   public boolean validOperands(int[ ] operands, int count) {
      boolean valid = true;
      for (int i = 0; i < count && valid; i++) {
         if (operands[i] == Integer.MIN_VALUE) valid = false;
      }
      return valid;
   }

   public void printInvalidOperands(int[ ] operands, int count) {
      for (int i = 0; i < count; i++) {
         if (operands[i] < 0) {
            System.out.print("Operand["+i+"], "+operands[i]);
            System.out.print(" is invalid");
         }
      }
      System.out.println(""); 
   }
}
